core/controllers/suggestion.py

```diff
@@ -148,6 +148,26 @@ def post(self):
 
         self.render_json(self.values)
 
+    # TODO(#11735): Currently, only handles the translation suggestion
+    # updates. This needs to update to generally handle the suggestions.
+    @acl_decorators.can_update_suggestions
+    def put(self, suggestion_id):
+        """Handles PUT requests.
+
+        Raises:
+            ValidationError. The suggestion is already handled.
+        """
+        suggestion = suggestion_services.get_suggestion_by_id(suggestion_id)
+        if suggestion.status == 'accepted':
+            raise utils.ValidationError(
+                'The suggestion with id %s has already been accepted'
+                % (suggestion_id)
+            )
+
+        suggestion_services.update_translation_suggestion(
+            suggestion_id, self.payload.get('translation_html'))
```

Comment:
validate the type of payload in the controller layer and raise InvalidInput if needed.

Reply:
Done

Reply:
I mean should we validate the type i.e, it's a string not integer or anything else.